tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/ResourcesSpec.scala

```diff
@@ -380,6 +381,27 @@ class ResourcesSpec extends BaseSpec with EitherValuable with CirceEq {
           filterMetadataKeys(json) should equalIgnoreArrayOrder(expected)
       }
     }
+
+    "allow to change the schema" in {
+      val resourceId       = "schemaChange"
+      val payload          = SimpleResource.sourcePayload(resourceId, 4)
+      val newSchemaPayload = SchemaPayload.loadSimpleNoId()
+
+      val postResourceAndNewSchema =
+        deltaClient.put[Json](s"/schemas/$id1/new-schema", newSchemaPayload, Rick) { expectCreated } >>
+          deltaClient.put[Json](s"/resources/$id1/test-schema/$resourceId", payload, Rick) { expectCreated } >>
+          deltaClient.get[Json](s"/resources/$id1/test-schema/$resourceId", Rick) { expectOk }
+
+      val updateResourceAndSchema =
+        deltaClient.put[Json](s"/resources/$id1/new-schema/$resourceId?rev=1", payload, Rick)(_)
+
+      postResourceAndNewSchema >>
+        updateResourceAndSchema { (json, response) =>
+          response.status shouldEqual StatusCodes.OK
+          _rev.getOption(json) should contain(2)
+          _constrainedBy.getOption(json) should contain("http://delta:8080/v1/resources/" + id1 + "/_/new-schema")
+        }
+    }
```

Comment:
It would be good to make it clear if you're updating or adding a schema to the resource. One way would be to label the payload something like `resourceWithASchemaPayload`

Or: 
```
thereIsASchema(schemaId) >>
  thereIsAResourceWithSchema(otherSchemaId) >>
  updateResourceSchema(resourceId, schemaId)
```

There's also a question of whether the functionality is actually tested here: when updating the schema the desired effect for the user is that now the resource is constrained by the schema. The fact that it's returning that it was constrained in the response doesn't actually show that it's constrained